charon/utils/archive.py

```diff
@@ -81,12 +87,33 @@ def _get_tgz_name(path: str):
     return ""
 
 
-def __parse_npm_package_version_paths(path: str) -> list:
+def _del_none(d):
+    for key, value in list(d.items()):
+        if value is None:
+            del d[key]
+        elif isinstance(value, dict):
+            _del_none(value)
+    return d
+
+
+def __write_npm_version_dist(path: str, version_meta_extract_path: str, version_data: dict,
+                             tgz_relative_path: str):
+    tarball_link = "".join(["https://", NRRC_REGISTRY, "/", tgz_relative_path])
+    shasum = digest(path, HashType.SHA1)
+    dist = dict()
+    dist["tarball"] = tarball_link
+    dist["shasum"] = shasum
```

Comment:
There is also a "integrity" item in this dist section, which has "sha512-xxxxxx" value.